File: 2015/ruby/day6/part1.rb
#!/usr/bin/ruby

begin
    File.open('input.txt', 'r') do |file|
        lights = Array.new(1000) { |light| Array.new(1000) { |light| false }}

        file.each_line do |line|
            words = line.split(' ')

            corner1 = words[words.length - 3].split(',')
            corner2 = words[words.length - 1].split(',')

            x1 = corner1[0].to_i
            y1 = corner1[1].to_i
            x2 = corner2[0].to_i
            y2 = corner2[1].to_i

            for row in y1..y2 do
                for col in x1..x2 do
                    if words[0] == 'toggle'
                        lights[row][col] = !lights[row][col]
                    elsif words[1] == 'on'
                        lights[row][col] = true
                    elsif words[1] == 'off'
                        lights[row][col] = false
                    end
                end
            end
        end

        total = 0
        for row in 0..999 do
            total += lights[row].count(true)
        end

        puts total
    end
rescue SystemCallError
    puts 'input.txt not found.'
end

File: 2016/python/day2/part1.py
#!/usr/bin/python

with open('input.txt') as file_handler:
    content = file_handler.readlines()

code = ''
button = 5

for line in content:
    for move in line.rstrip():
        if move == 'U' and button > 3:
            button -= 3
        elif move == 'R' and button % 3 != 0:
            button += 1
        elif move == 'D' and button < 7:
            button += 3
        elif move == 'L' and button % 3 != 1:
            button -= 1

    code += str(button)

print code
